### DISCUSSION TITLE
The route log-viewer/api/folders could not be found

### DISCUSSION THREAD

**Discussion by entoniperez** [opcodesio/log-viewer#280] [Q&A]
From my local machine, I get this error when I try to get remote production log:

![image](https://github.com/opcodesio/log-viewer/assets/12221202/6ce45226-b57f-4ecd-a639-f9fd9fa7b90f)

The project on production is in a subdomain in a shared hosting. And, if I try my website link `https://subdomain.mydomain.com/log-viewer` it returns a `404 | NOT FOUND` Laravel view. But it works locally...

I'm missing something?

**Comment by arukompas** [comment]
hey @entoniperez , your first mission is to figure out why the https://subdomain.mydomain.com/log-viewer route is not working. Perhaps there's some misconfiguration on the subdomain or shared hosting side. I don't have experience with shared hosting providers, so I can't consult on that.

Try seeing if other routes of your app (not just the home page, but other pages too) work as expected, because I sense this might be an issue on the whole site, not just Log Viewer.

Let me know what you find.

**Reply by entoniperez** [reply]
Other routes works as expected... it's really strange. I was thinking that maybe I'm missing something about authorizing access to Log Viewer, but if I didn't change anything after installation, then on production should works as in local, right? Thanks for your time.

**Reply by arukompas** [reply]
@entoniperez could it be that your routes were cached in production? If they were cached before you installed Log Viewer and never re-cached again, that might explain it.

Run `php artisan route:cache` in production.

**Reply by entoniperez** [reply]
I've tried it calling this url through a route in web.php file:

```php
Route::get('route-clear-cache', function () {
    Artisan::call('route:clear');
    Artisan::call('route:cache');
    return "Routes has been cleared and cached again.";
});
```

Same result for the url **/log-viewer**: 404.

**Comment by christoferd** [comment]
Suddenly having the same issue on production server.
It worked after I deleted the file: /config/log-viewer.php

**Reply by arukompas** [reply]
@christoferd did you at some point upgrade from an older Log Viewer version? I wonder if the config changes were clashing and something got messed up. Or maybe you had defined a different path for the log viewer? I.e. instead of `/log-viewer` you defined a different path to access it?

**Reply by christoferd** [reply]
@arukompas Not sure, but it's possible that I was missing something in the config, sometimes I run composer update.

**Comment by jcerdan** [comment]
Hi, I have exactly the same problem.
Created again config file did not solve the issue
Furthermore, the route is shown when I run ```php artisan route:list```
And I can access log-viewer directly.

My explain my situation as I think it can give some clues. It's as follows:

I have a website logs.webs.com (as example).  This website offers access to several logs on other websites: 
website-a.com, 
website-b.com, 
website-c.com, 
etc...

All websites work ok and I can access their log files via logs.webs.com
The only one that gives an error is the website hosted on the same server (not-working.com). All other websites are hosted on other servers.
I can access the logs of not-working.com if I go to the URL directly (not via logs.webs.com), so log-viewer works fine there.
I think that, somehow in name resolution, it detects the website is in the same host and so it tries to access the route on logs.webs.com instead of not-working.com, and then it results in route not found.

Can this throw some light on the issue?

**Reply by arukompas** [reply]
hey @jcerdan , when defining the host in `config/log-viewer.php` for the "not-working.com" website, do you use `'host' => null` (as in, local), or do you still set it to `not-working.com/log-viewer` ?

From the server that hosts logs.webs.com, can you "ping not-working.com" ? does it resolve everything correctly?
Can you visit `https://not-working.com/log-viewer` in the browser?
What is the contents of `config/log-viewer.php` for the not-working website?

**Reply by jcerdan** [reply]
Hi @arukompas 

in logs.webs.com I have set

```
    'not-working' => [
            'name' => 'not-working',
            'host' => 'https://www.not-working.com/log-viewer',
        ],
```

Yes, from logs.webs.com I can ping not-working.com and it works fine.
Also yes, I can visit ```https://not-working.com/log-viewer``` from Firefox directly.
Also ```LogViewer::auth``` function allows access to everyone (it just returns true)

I attach config file in ```not-working.com```

[log-viewer.txt](https://github.com/user-attachments/files/18428626/log-viewer.txt)
